File: question/neetcode150/two_pointer/two-sum-ii-input-array-is-sorted.py
from typing import List

class Solution:
    def twoSum(self, a: List[int], t: int) -> List[int]:
        l = 0
        r = len(a)-1

        while True:
            if a[l]+a[r] > t:
                r-=1
            elif a[l]+a[r] < t:
                l+=1
            else:
                return l+1,r+1

File: question/neetcode150/two_pointer/two-sum-ii-input-array-is-sorted.cpp
#include <bits/stdc++.h>
 
using namespace std;

class Solution {
public:
    vector<int> twoSum(vector<int>& a, int t) {
        int n = a.size();
        int l = 0, r = n-1;

        while (1){
            if(a[l]+a[r] == t){
                return {l+1, r+1};
            }
            else if (a[l]+a[r] < t){
                l++;
            }
            else
                r--;
        }
    }
};